compiler/rustc_codegen_llvm/src/abi.rs

```diff
@@ -415,9 +415,36 @@ impl<'ll, 'tcx> FnAbiLlvmExt<'ll, 'tcx> for FnAbi<'tcx, Ty<'tcx>> {
             i += 1;
             i - 1
         };
+
+        let apply_range_attr = |idx: AttributePlace, abi: rustc_target::abi::Abi| {
+            if cx.sess().opts.optimize != config::OptLevel::No
+                && llvm_util::get_version() >= (19, 0, 0)
+            {
+                if let abi::Abi::Scalar(scalar) = abi {
+                    // If the value is a boolean, the range is 0..2 and that ultimately
+                    // become 0..0 when the type becomes i1, which would be rejected
+                    // by the LLVM verifier.
+                    if let Int(..) = scalar.primitive() {
+                        if !scalar.is_bool() && !scalar.is_always_valid(cx) {
```

Comment:
Don't quote me on this but I think it is okay to use let chains in the compiler to avoid all this nesting

```suggestion
        let apply_range_attr = |idx: AttributePlace, abi: rustc_target::abi::Abi| {
            if cx.sess().opts.optimize != config::OptLevel::No
                && llvm_util::get_version() >= (19, 0, 0)
                && let abi::Abi::Scalar(scalar) = abi
                // If the value is a boolean, the range is 0..2 and that ultimately
                // become 0..0 when the type becomes i1, which would be rejected
                // by the LLVM verifier.
                && matches!(scalar.primitive(), Int(..))
                && !scalar.is_bool() && !scalar.is_always_valid(cx) {
```

Reply:
Yes please

Reply:
One day stable :(

Reply:
nice did not think it was possible to do that.